src/libertem/analysis/com.py

```diff
@@ -154,6 +172,84 @@ def magnitude(y_centers, x_centers):
+        for flip_y in (True, False):
+            y_transformed, x_transformed = apply_correction(
+                y_centers, x_centers, scan_rotation=angle, flip_y=flip_y
+            )
+            curl = curl_2d(y_transformed, x_transformed)
+            # The last row and column contain artefacts
+            result = np.sqrt(np.mean(curl[:-1, :-1]**2))
+            if flip_y:
+                flipped[angle] = result
+            else:
+                straight[angle] = result
+    return (straight, flipped)
+
+
+def guess_corrections(y_centers, x_centers):
+    '''
+    Guess corrections for center shift, :code:`scan_rotation` and :code:`flip_y` from CoM data
+
+    This function can generate a CoM parameter guess for atomic resolution 4D STEM data
+    by using the following assumptions:
+
+    * The field is purely electrostatic, i.e. the RMS curl should be minimized
+    * There is no net field over the field of view and no descan error,
+      i.e. the mean deflection is zero.
+    * Atomic resolution STEM means that the divergence will be negative at atom columns
+      and consequently the histogram of divergence will have a stronger tail towards
+      negative values than towards positive values.
+
+    If any corrections were applied when generating the input data, please note that the corrections
+    should be applied relative to these previous value. In particular, the
+    center corrections have to be back-transformed, for example with
+    :code:`apply_corrections(..., forward=False)
+
+    Returns
+    -------
+    scan_rotation, flip_y, cy, cx : relative to current values
+    '''
+    straight, flipped = coordinate_check(y_centers, x_centers)
+    # The one with lower minima is the correct one
+    flip_y = bool(np.min(flipped) < np.min(straight))
+    if flip_y:
+        angle = np.argmin(flipped)
+    else:
+        angle = np.argmin(straight)
+
+    corrected_y, corrected_x = apply_correction(
+        y_centers, x_centers, scan_rotation=angle, flip_y=flip_y
+    )
+    # There are two equivalent angles that minimize RMS curl since a 180°
+    # rotation inverts the coordinates and just flips the sign for divergence
+    # and curl. To distinguish the two, the distribution of the divergence is
+    # analyzed. With negative electrons and positive nuclei, the beam is
+    # deflected towards the nuclei and the divergence is negative there. Since
+    # the beam is deflected most strongly near the nuclei, the histogram should
+    # have more values at the negative end of the range than at the positive
+    # end.
+    div = divergence(corrected_y, corrected_x)
+    all_range = np.maximum(-np.min(div), np.max(div))
+    hist, bins = np.histogram(div, range=(-all_range, all_range), bins=5)
+    polarity_off = np.sum(hist[:1]) < np.sum(hist[-1:])
+    if polarity_off:
+        angle += 180
+    if angle > 180:
+        angle -= 360
+    return (angle, flip_y, np.mean(y_centers), np.mean(x_centers))
```

Comment:
Maybe we could use a `namedtuple` here instead? That would minimize errors from messing up the ordering (y/x vs x/y) when unpacking (and still allow unpacking like a normal tuple, if you really want that)